docs/decisions/0011-emit-course-completion-events.rst

```diff
@@ -0,0 +1,30 @@
+Emit course completion events from edX platform
+===============================================
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+#. edX's implementation of xAPI emits course completion event when learner passes a course. Having passed the course once, learners can still fail and pass the same course if they attempt the course content again or if the grading criteria changes.
+
+#. edX's implementation of xAPI ensures that course completion event for a specific learner and course is emitted only once i.e. the first time this learner passes the course.
```

Comment:
Why does this have to be true?  I don't understand where the emit at-most once requirement is coming from.  Can you explain why the current situation of multiple emission is a problem?

Reply:
We want our event consumers to have both options:
1. Track a course completion event that is emitted only once, i.e. the first time a learner passes a course.
2. Track a course completion event that is emitted every time the learner achieves passing grade.
They can achieve this by subscribing to edx.course.completed and edx.course.grade.now_passed events respectively. Also, edx.course.completed has now been renamed to course_grade_passed_first_time.

Reply:
@Ayub-Khan it makes sense to emit both of these events.